From 44d3faded15eecfc9662007bc7ccca9f959c64d0 Mon Sep 17 00:00:00 2001
Message-ID: <44d3faded15eecfc9662007bc7ccca9f959c64d0.1705743628.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 17 Jan 2024 22:13:38 +0000
Subject: [PATCH 35/38] gnu: go-github.com-smartystreets-assertions: Move to
 (gnu packages golang-check).

* gnu/packages/check.scm (go-github.com-smartystreets-assertions): Move
from here ...
* gnu/packages/golang-check.scm: ... to here.

Change-Id: Ia90bbe984fe19dd48d9df91364a7ecf2bf8bd37f
---
 gnu/packages/check.scm        | 33 ---------------------------------
 gnu/packages/golang-check.scm | 33 +++++++++++++++++++++++++++++++++
 2 files changed, 33 insertions(+), 33 deletions(-)

diff --git a/gnu/packages/check.scm b/gnu/packages/check.scm
index 0b7acaddad..5e82a09126 100644
--- a/gnu/packages/check.scm
+++ b/gnu/packages/check.scm
@@ -898,39 +898,6 @@ (define-public python-gixy
 command.")
       (license license:mpl2.0))))
 
-(define-public go-github.com-smartystreets-assertions
-  (package
-    (name "go-github.com-smartystreets-assertions")
-    (version "1.13.0")
-    (source (origin
-              (method git-fetch)
-              (uri (git-reference
-                    (url "https://github.com/smartystreets/assertions")
-                    (commit (string-append "v" version))))
-              (file-name (git-file-name name version))
-              (sha256
-               (base32 "0flf3fb6fsw3bk1viva0fzrzw87djaj1mqvrx2gzg1ssn7xzfrzr"))))
-    (build-system go-build-system)
-    (arguments
-     (list
-       #:import-path "github.com/smartystreets/assertions"
-       #:phases
-       #~(modify-phases %standard-phases
-           (replace 'check
-             (lambda* (#:key inputs #:allow-other-keys #:rest args)
-               (unless
-                 ;; The tests fail when run with gccgo.
-                 (false-if-exception (search-input-file inputs "/bin/gccgo"))
-                 (apply (assoc-ref %standard-phases 'check) args)))))))
-    (native-inputs
-     (list go-github.com-smartystreets-gunit))
-    (synopsis "Assertions for testing with Go")
-    (description
-     "The @code{assertions} package provides convenient assertion functions
-for writing tests in Go.")
-    (home-page "https://github.com/smartystreets/assertions")
-    (license license:expat)))
-
 (define-public go-github.com-smartystreets-goconvey
   (package
     (name "go-github.com-smartystreets-goconvey")
diff --git a/gnu/packages/golang-check.scm b/gnu/packages/golang-check.scm
index 207e4a548f..0c37849806 100644
--- a/gnu/packages/golang-check.scm
+++ b/gnu/packages/golang-check.scm
@@ -4,6 +4,7 @@
 ;;; Copyright © 2018 Pierre-Antoine Rouby <pierre-antoine.rouby@inria.fr>
 ;;; Copyright © 2019 Brian Leung <bkleung89@gmail.com>
 ;;; Copyright © 2019 Leo Famulari <leo@famulari.name>
+;;; Copyright © 2019 Vagrant Cascadian <vagrant@debian.org>
 ;;; Copyright © 2020 Jakub Kądziołka <kuba@kadziolka.net>
 ;;; Copyright © 2020 Joseph LaFreniere <joseph@lafreniere.xyz>
 ;;; Copyright © 2020 Oleg Pykhalov <go.wigust@gmail.com>
@@ -379,6 +380,38 @@ (define-public go-github-com-onsi-ginkgo
 Gomega matcher library.")
     (license license:expat)))
 
+(define-public go-github.com-smartystreets-assertions
+  (package
+    (name "go-github.com-smartystreets-assertions")
+    (version "1.13.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/smartystreets/assertions")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256 (base32 "0flf3fb6fsw3bk1viva0fzrzw87djaj1mqvrx2gzg1ssn7xzfrzr"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/smartystreets/assertions"
+      #:phases
+      #~(modify-phases %standard-phases
+          (replace 'check
+            (lambda* (#:key inputs #:allow-other-keys #:rest args)
+              (unless
+                  ;; The tests fail when run with gccgo.
+                  (false-if-exception (search-input-file inputs "/bin/gccgo"))
+                (apply (assoc-ref %standard-phases 'check) args)))))))
+    (native-inputs
+     (list go-github.com-smartystreets-gunit))
+    (home-page "https://github.com/smartystreets/assertions")
+    (synopsis "Assertions for testing with Go")
+    (description "The @code{assertions} package provides convenient assertion
+functions for writing tests in Go.")
+    (license license:expat)))
+
 (define-public go-github.com-smartystreets-gunit
   (package
     (name "go-github.com-smartystreets-gunit")
-- 
2.41.0

